rasa/utils/plotting.py

```diff
@@ -59,7 +59,7 @@ def _needs_matplotlib_backend(func: FuncType) -> FuncType:
     """Decorator to fix matplotlib backend before calling a function."""
 
     @wraps(func)
-    def inner(*args: Any, **kwargs: Any) -> ReturnType:
+    def inner(*args: Any, **kwargs: Union[Any, ReturnType]) -> ReturnType:
```

Comment:
have you tried using the new [`ParamSpec`](https://docs.python.org/3/library/typing.html#typing.ParamSpec)?

Reply:
> expected to see at least one of the function arguments typed to the same type as the return type

this is weird though - sounds like a mypy error

Reply:
Should I mark with a type ignore comment?

Reply:
Addressed: https://github.com/RasaHQ/rasa/pull/11773/commits/bfd5135e0d7127da726a92b50602c29ae1991636